app/src/main/java/io/gnosis/safe/ui/transaction/TransactionsViewModel.kt

```diff
@@ -1,103 +1,122 @@
 package io.gnosis.safe.ui.transaction
 
 import androidx.annotation.StringRes
-import io.gnosis.data.models.Page
+import androidx.lifecycle.viewModelScope
+import androidx.paging.PagingData
+import androidx.paging.cachedIn
+import androidx.paging.insertSeparators
 import io.gnosis.data.models.Safe
 import io.gnosis.data.models.Transaction
 import io.gnosis.data.models.TransactionStatus
 import io.gnosis.data.repositories.SafeRepository
-import io.gnosis.data.repositories.TransactionRepository
 import io.gnosis.safe.R
 import io.gnosis.safe.ui.base.AppDispatchers
 import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.transaction.list.TransactionPagingRepository
+import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.map
 import pm.gnosis.model.Solidity
 import javax.inject.Inject
 
 class TransactionsViewModel
 @Inject constructor(
-    private val transactionRepository: TransactionRepository,
+    private val transactionsPagingRepository: TransactionPagingRepository,
     private val safeRepository: SafeRepository,
     appDispatchers: AppDispatchers
 ) : BaseStateViewModel<TransactionsViewState>(appDispatchers) {
 
+    private var currentSafeAddress: Solidity.Address? = null
```

Comment:
I could be wrong, but judging by where you use this `var` it might not be necessary at all to have it. I could have missed a place though.

Reply:
it is used for caching the paging results in the viewModel scope

Reply:
Couldn't you use `safeRepository.getActiveSafe` every time you need the current safe instead of caching? That is technically the only source of truth you should trust when it comes to active safe.

Reply:
we need to save last safe address. Otherwise we would not know if safe was changed when we select another safe from  safe selection dialog.

Reply:
Don't we have a `flow` in the `SafeRepository` that we could use for this?